java/java.file.launcher/src/org/netbeans/modules/java/file/launcher/SingleSourceFileUtil.java

```diff
@@ -82,10 +82,18 @@ public static boolean isSingleSourceFile(FileObject fObj) {
     }
 
     public static boolean isSupportedFile(FileObject file) {
+        if (file == null) {
+            return false;
+        }
         try {
-            return !MultiSourceRootProvider.DISABLE_MULTI_SOURCE_ROOT &&
-                   FileOwnerQuery.getOwner(file) == null &&
-                   !file.getFileSystem().isReadOnly();
+            FileObject dir = file.getParent();
+            File dirFile = dir != null ? FileUtil.toFile(dir) : null;
+            return !MultiSourceRootProvider.DISABLE_MULTI_SOURCE_ROOT
+                    && FileOwnerQuery.getOwner(file) == null
+                    && !file.getFileSystem().isReadOnly()
+                    && !(dirFile != null
+                    && dirFile.getName().startsWith("vcs-")
+                    && dirFile.getAbsolutePath().startsWith(System.getProperty("java.io.tmpdir")));
```

Comment:
nitpick: this growing chain of seemingly unrelated conditions might look better when torn apart into multiple checks:

```java
if (file == null || MultiSourceRootProvider.DISABLE_MULTI_SOURCE_ROOT) {
    return false;
}

try {
    if (fileConditionsFail)
        return false,
    }
    File dirFile = ...
    if (parentConditionsFail) {
        return false;
    }
    return true;

```